genvm/vm.go

```diff
@@ -291,18 +292,18 @@ func (v *VM) execute(lctx ApplyContext, ss *core.StagedCache, txs []types.Transa
 		}
 		ctx := req.ctx
 		args := req.args
-
-		if ctx.Account.Balance < ctx.Header.MaxGas*ctx.Header.GasPrice {
-			v.logger.With().Warning("ineffective transaction. can't cover gas and max spend",
+		if ctx.Account.Balance < ctx.ParseOutput.FixedGas*ctx.ParseOutput.GasPrice {
```

Comment:
why isn't this compared to ctx.Header.MaxGas * ctx.ParseOutput.GasPrice?
i.e. we should also consider if the principal can afford the storage cost too

Reply:
> if account can cover fixed gas, but can't cover max gas tx will be consumed and failed with out of balance.

this is the other part of this pr. the gas is consumed in ctx.Consume before ctx.Exec

Reply:
right. so checking it here confuses me. 

theoretically `ctx.Account.Balance < ctx.ParseOutput.FixedGas*ctx.ParseOutput.GasPrice` cannot be true because the code already checks if the balance can cover max_gas*gas_price

Reply:
it can be true. fixed gas is lower than max gas

if account can't cover fixed gas - tx ineffective
if account gas is [fixed gas, max gas) - tx will be failed and remaining balance is consumed
if account gas [max gas, ...) - tx is executed

Reply:
yikes. i confused execute() with handler.Exec(). sorry. make sense. thanks